src/main/java/com/redhat/rhjmc/containerjfr/net/web/http/api/v1/TargetRecordingPatchSave.java

```diff
@@ -116,7 +120,25 @@ private String saveRecording(JFRConnection connection, IRecordingDescriptor desc
         if (recordingName.endsWith(".jfr")) {
             recordingName = recordingName.substring(0, recordingName.length() - 4);
         }
-        String targetName = connection.getHost().replaceAll("[\\._]+", "-");
+
+        String targetName =
+                platformClient.listDiscoverableServices().stream()
```

Comment:
This is my only issue - it's unfortunate that this lookup is performed every time. We should implement something that allows us to do this cross-referencing without having to trigger target discovery first. Something as simple as a map from targetIds to known ServiceRefs would fit the description, but I'm not sure if that data structure should be held by the PlatformClient or if we should abstract it somewhere else. Maybe just leave a TODO comment here with some short summary and description of this problem. It could also be filed as an issue of type "chore".